modules/core/src/v2/coins/trx.ts

```diff
@@ -249,6 +275,200 @@ export class Trx extends BaseCoin {
+   */
+  getBuildTransaction(toAddr: string, fromAddr: string, amount: number, callback?: NodeCallback<any>): Bluebird<any> {
+    const self = this;
+    return co(function*() {
+      // our addresses should be base58, we'll have to encode to hex
+      const result = yield self.recoveryPost({
+        path: '/wallet/createtransaction',
+        jsonObj: {
+          to_address: toAddr,
+          from_address: fromAddr,
+          amount: amount.toFixed(0),
+        },
+        node: NodeTypes.Full,
+      });
+      return result.balance;
+    })
+        .call(this)
+        .asCallback(callback);
+  }
+
+  /**
+   * Builds a funds recovery transaction without BitGo.
+   * We need to do three queries during this:
+   * 1) Node query - how much money is in the account
+   * 2) Build transaction - build our transaction for the amount
+   * 3) Send signed build - send our signed build to a public node
+   * @param params
+   * @param callback
+   */
+  recover(params: RecoveryOptions, callback?: NodeCallback<RecoveryTransaction>): Bluebird<RecoveryTransaction> {
+    const self = this;
+    return co<RecoveryTransaction>(function*() {
+      const isKrsRecovery = params.backupKey.startsWith('xpub') && !params.userKey.startsWith('xpub');
+      const isUnsignedSweep = params.backupKey.startsWith('xpub') && params.userKey.startsWith('xpub');
+
+      // get our user, backup keys
+      const keys = yield self.initiateRecovery(params);
+
+      // we need to decode our bitgoKey to a base58 address
+      const bitgoAddress = self.compressedPubToHexAddress(self.xpubToCompressedPub(params.bitgoKey));
+      const recoveryAddressHex = bitgoAccountLib.Trx.Utils.getHexAddressFromBase58Address(params.recoveryDestination);
+
+      const recoveryAmount = yield self.getAccountBalanceFromNode(bitgoAddress);
+
+      //const userPrv = HDKey.fromExtendedKey(keys[0]).privateKey;
+      const userXPub = keys[0].neutered().toBase58();
+      const userXPrv = keys[0].toBase58();
+      const backupXPub = keys[1].neutered().toBase58();
+
+      const userPrv = self.xprvToCompressedPrv(userXPrv);
+      const userHexAddr = self.compressedPubToHexAddress(self.xpubToCompressedPub(userXPub));
+      const backupHexAddr = self.compressedPubToHexAddress(self.xpubToCompressedPub(backupXPub));
+
+      // TODO: some checks here about pubs being valid, for this wallet, etc.
+
+      // construct the tx
+      // TODO: recovery amount might not include fees, we'll have to figure out the max we can spend
+      //   sensitive to energy and tx cost amounts
+      // there's an assumption here being made about fees: for a wallet that hasn't been used in awhile, the implication is
+      // it has maximum bandwidth. thus, a recovery should cost the minimum amount (1e6 sun)
+      if (1e6 > recoveryAmount) {
+        throw new Error('Amount of funds to recover wouldnt be able to fund a send');
+      }
+      const recoveryAmountMinusFees = recoveryAmount - 1e6;
```

Comment:
Are we sure this is safe to hard code? Might this be changed in the future?

Reply:
could be. I added a const for it. in theory should be sourced from account lib @argjv